### ISSUE TITLE
parsing_node_ids blows up when a non-last node id in a list of passed node ids fails to parse

### ISSUE THREAD

**Issue by dpehrson** [absinthe-graphql/absinthe_relay#49]
Given a resolver using `parsing_node_ids` for multiple args as such:

`resolve parsing_node_ids(&my_resolver/2, id1: :foo, id2: foo)`

If any of the non-last arguments (here, `id1`) fail to parse an exception is thrown:

`** (BadMapError) expected a map, got: {:error, "Invalid node type for argument id1; should be foo, was other_foo"}`

I believe this is because the value of the `args` variable in the implementation of `parsing_node_ids` is used for multiple purposes within the function and upon the first error within the `Enum.reduce` the original args are replaced with the error tuple from the first parsing error causing any subsequent calls to `Map.get` to fail.

- [My fork branch with expanded tests, including one that shows this error](https://github.com/dpehrson/absinthe_relay/tree/parsing-node-ids)
- [A comparison of that branch against master](https://github.com/absinthe-graphql/absinthe_relay/compare/master...dpehrson:parsing-node-ids)

And finally a replacement `node_test.exs` that can be used to illustrate the problem with additional (passing) tests around `parsing_node_ids`:

```elixir
defmodule Absinthe.Relay.NodeTest do
  use Absinthe.Relay.Case, async: true

  alias Absinthe.Relay.Node

  defmodule Schema do
    use Absinthe.Schema
    use Absinthe.Relay.Schema

    @foos %{
      "1" => %{id: "1", name: "Bar 1"},
      "2" => %{id: "2", name: "Bar 2"}
    }

    node interface do
      resolve_type fn
        _, _  ->
          # We just resolve :foos for now
          :foo
      end
    end

    node object :foo do
      field :name, :string
    end

    node object :other_foo, name: "FancyFoo" do
      field :name, :string
    end

    query do
      field :single_foo, :foo do
        arg :id, non_null(:id)
        resolve parsing_node_ids(&resolve_foo/2, id: :foo)
      end

      field :dual_foo, list_of(:foo) do
        arg :id1, non_null(:id)
        arg :id2, non_null(:id)
        resolve parsing_node_ids(&resolve_foos/2, id1: :foo, id2: :foo)
      end
    end

    defp resolve_foo({:error, _msg} = error, _info), do: error
    defp resolve_foo(%{id: id}, _) do
      {:ok, Map.get(@foos, id)}
    end

    defp resolve_foos(%{id1: id1, id2: id2}, _) do
      {:ok, [
        Map.get(@foos, id1),
        Map.get(@foos, id2)
      ]}
    end

  end

  @foo1_id Base.encode64("Foo:1")
  @foo2_id Base.encode64("Foo:2")

  describe "to_global_id" do

    it "works given an atom for an existing type" do
      assert !is_nil(Node.to_global_id(:foo, 1, Schema))
    end

    it "returns an atom for an non-existing type" do
      assert is_nil(Node.to_global_id(:not_foo, 1, Schema))
    end

    it "works given a binary and internal ID" do
      assert Node.to_global_id("Foo", 1)
    end

    it "gives the same global ID for different type, equivalent references" do
      assert Node.to_global_id("FancyFoo", 1) == Node.to_global_id(:other_foo, 1, Schema)
    end

    it "gives the different global ID for different type, equivalent references" do
      assert Node.to_global_id("FancyFoo", 1) != Node.to_global_id(:foo, 1, Schema)
    end

    it "fails given a bad ID" do
      assert is_nil(Node.to_global_id("Foo", nil))
    end

  end

  describe "parsing_node_id_args" do

    it "parses one id correctly" do
      result = """
      { singleFoo(id: "#{@foo1_id}") { id name } }
      """ |> Absinthe.run(Schema)
      assert {:ok, %{data: %{"singleFoo" => %{"name" => "Bar 1", "id" => @foo1_id}}}} == result
    end

    it "handles one incorrect id" do
      result = """
      { singleFoo(id: "#{Node.to_global_id(:other_foo, 1, Schema)}") { id name } }
      """ |> Absinthe.run(Schema)
      assert {:ok, %{data: %{}, errors: [
        %{message: "In field \"singleFoo\": Invalid node type for argument `id`; should be foo, was other_foo"}
      ]}} = result
    end

    it "parses multiple ids correctly" do
      result = """
      { dualFoo(id1: "#{@foo1_id}", id2: "#{@foo2_id}") { id name } }
      """ |> Absinthe.run(Schema)
      assert {:ok, %{data: %{"dualFoo" => [
        %{"name" => "Bar 1", "id" => @foo1_id},
        %{"name" => "Bar 2", "id" => @foo2_id}
      ]}}} == result
    end

    # This never succeeeds.
    # The current implementation of `parsing_node_ids` clobbers the `args` variable on the first failure of `id1`
    # causing the next iteration of the `Enum.reduce` when processing `id2` to blow up when it tries to `Map.get`
    # on the now-clobbered `args` that is actually now a `{:error, ...}` tuple resulting from `id1`
    it "handles multiple incorrect ids" do
      result = """
      { dualFoo(id1: "#{Node.to_global_id(:other_foo, 1, Schema)}", id2: "#{Node.to_global_id(:other_foo, 2, Schema)}") { id name } }
      """ |> Absinthe.run(Schema)
      assert {:ok, %{data: %{}, errors: [
        %{message: "In field \"multipleFoo\": Invalid node type for argument `id1`; should be foo, was other_foo"},
        %{message: "In field \"multipleFoo\": Invalid node type for argument `id2`; should be foo, was other_foo"}
      ]}} = result
    end

  end

end
```

**Comment by dpehrson** [comment]
Unfortunately I am not clear as to what the intended behavior of `parsing_node_ids` is in these situations so I am not totally sure what the fix would be.

**Comment by bruce** [comment]
Fix released in v1.3.0-beta.1; your test coverage aided significantly (I cherry-picked it in). Thanks!
